### PR TITLE
Meta+Ports: You shall no longer make questionable patches

### PR DESCRIPTION
**Pull request by alimpfard**
Allow me to set a scene:
- Go take a look at `ls Ports/*/patches`, go on, I'll wait.
- Can you pick any one of them and tell what it does? OF COURSE NOT!
- So you sit down and read the damn thing, then question the original author's sanity for a while, then question your own sanity for a while, then give up and go chase another yak.

And now, allow me to present a solution (in multiple parts):
- First, make it illegal to not explain your patches (this PR)
- Then, give people some nice tools to make nice, tidy atomic patches - none of this `Ports/foo/patches/foo.patch` shit anymore
- Then, allow autogenerating the ReadMe files from the patches (as they're theoretically nice, and made by git!)
- Then, gradually replace the ones missing from this PR as ports are renewed

PROBLEM SOLVED, THANKS FOR COMING TO MY TED TALK.

### CODE REVIEW

**Comment by IdanHo** [comment]
commit 2 and commit 3 should probably be swapped, since we theoretically want to the system to pass all tests in every commit

**Comment by alimpfard** [comment]
The linters aren't run as tests, but sure, I'll swap them

**Comment by bgianfo** [comment]
> * Then, allow autogenerating the ReadMe files from the patches

Can we do this part now, instead of checking in useless `ReadMe.md` files? We already have some self documenting git based patches in the Ports tree.

**Comment by alimpfard** [comment]
I believe I grabbed the descriptions from patches that were based on git, so the only ones with FIXME contents should be ones that don't have any of that.
But I can certainly make a script that does that and put it in the PR.

File: Ports/stress-ng/patches/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Patches for stress-ng
+
+## `0010-disable-rand48-cpu-stressor.patch`
+FIXME: This patch should be documented, and possibly split into multiple atomic patches.
```
Comment (alimpfard) [inline Ports/stress-ng/patches/ReadMe.md:4]: Looks like I missed at least this one, I'll make that script and update this...not today though.

**Comment by alimpfard** [comment]
(Still not complete)

New changes:
- Instead of checking in useless ReadMe.md files, just add a ignore list to the linter
- Introduce `./package.sh dev`, it does some magic and makes you a git environment in the apparent workdir, and spits out patches if you have commits in the _real_ one (can be pushed to with `git push`)
- Introduce `./package.sh generate_patch_readme` that generates a ReadMe.md from your patches

Things left to iron out still:
- Integration with the rest of `./package.sh`: Currently only building with it is supported
- Figure out a standard-ish commit structure for the patches and validate it as best we can.

**Comment by alimpfard** [comment]
I think this is just about done.
This checks out _all_ the original goals of this PR.

The "suggested" workflow for new ports (at least by me) is now:
- Write a small `package.sh` that includes `port`, `files` and optionally `depends` (at least)
- Run `./package.sh dev`, wait for it to do its magic
- `cd` into the workdir
- Make it build however you want, note that `./package.sh` inside the dev environment will operate on whatever you've `git push`'d and then reset it after building.
- Once everything is done, quit the dev shell, which will generate patches if needed and prompt you for whether you want to generate a ReadMe.

"Old" ports can relatively easily be imported by simply running `./package.sh dev` and going through the _wizard_


Note that it's _highly_ encouraged to produce small, atomic commits (and so patches), with decent explanations so we can remove them or replace them easily later when such changes are needed.

File: Ports/stress-ng/patches/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Subject: serenity: Disable rand48 cpu stressor, it's not implemented for Serenity
+
+## `0008-serenity-Make-lsearch-stressor-a-nop-on-Serenity.patch`
+
+Subject: serenity: Make lsearch stressor a nop on Serenity
+
+## `0009-serenity-Fake-O_SYNC-for-serenity-so-iomix-stressor-.patch`
+
+Subject: serenity: Fake O_SYNC for serenity so iomix stressor compiles
```
Comment (bgianfo) [inline Ports/stress-ng/patches/ReadMe.md:43]: It looks like the other generated `ReadMe.md`'s don't include the `Subject:` prefix, but this one does for some reason. Do you know what's up with that?

Comment (alimpfard) [reply Ports/stress-ng/patches/ReadMe.md:43]: oh I guess the regex is wrong?
I'll take a look :thonk:

Comment (alimpfard) [reply Ports/stress-ng/patches/ReadMe.md]: Fixed, I had updated the regex without regenerating these.

File: Ports/python3/patches/0001-Enforce-UTF-8-as-the-locale-encoding.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
+From ae8a6975f4fc63832421960e2766b9444edefe38 Mon Sep 17 00:00:00 2001
+From: Ali Mohammad Pur <ali.mpfard@gmail.com>
```
Comment (alimpfard) [inline Ports/python3/patches/0001-Enforce-UTF-8-as-the-locale-encoding.patch:2]: Gee Linus, tell me that I forgot to commit these with the correct name...

**Pull request merged**